lib/src/StepTracker/steplabel.scss

```diff
@@ -0,0 +1,122 @@
+@use "sass:math";
+@import "~@sebgroup/bootstrap/scss/variables";
+
+$step-label-component-name: "step-label";
+$step-label-label-element-name: "__label";
+$step-label-name-element-name: "__name";
+$step-number-height: 20px;
+$step-number-width: $step-number-height;
+$step-line: 4px;
+$step-line-height: 100%;
+
+@mixin tracker-line() {
+    &:not(:last-of-type) {
+        &::after {
+            @content;
+        }
+    }
+}
+
+.#{$step-label-component-name} {
```

Comment:
maybe we can utilise the ampersand trick to control the nesting of BEM naming convention
https://css-tricks.com/using-sass-control-scope-bem-naming/#so-how-does-this-relate-to-sass

```scss
.step-label {
    $step: &;

    display: flex;
    flex: 1;
    position: relative;
    gap: 0.5rem;
    align-items: center;

    @include tracker-line() {
        content: "";
        position: absolute;
        height: $step-line-height;
        top: calc(50% + #{math.div($step-number-height, 2)});
        left: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
        width: $step-line;
        background: linear-gradient(to left, $gray-300 50%, $blue-darker 50%) right;
        background-size: 300% 100%;
        background-position: right;
        transition: background-position 200ms ease-out;
    }

    &:hover {
        cursor: pointer;
    }

    &__label {
        flex: 0 0 $step-number-width;
        border-radius: 50%;
        border: 2px solid $gray-300;
        background-color: $gray-300;
        height: $step-number-height;
        width: $step-number-width;
        text-align: center;

        &--numbered {
            line-height: unset;
            font-size: $font-size-xs;
        }
    }

    &__name {
        position: relative;
        font-size: $font-size-sm;
        flex: 1;
        line-height: $step-number-height;
    }

    &--active {
        #{step}__label {
            border-color: $blue-darker;
        }

        #{$step}__name {
            color: $blue-darker;
        }
    }

    &--completed {
        @include tracker-line() {
            background-position: left;
        }

        #{$step}__label {
            color: $white;
            border-color: $blue-darker;
            background-color: $blue-darker;

            > svg {
                width: $font-size-xs;
                height: $font-size-xs;
                vertical-align: $step-line;
            }
        }
    }

    &--left {
        flex-direction: row-reverse;

        @include tracker-line() {
            left: unset;
            right: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
        }
    }

    &--bottom,
    &--top {
        flex-direction: column;
        padding-bottom: 0;

        @include tracker-line() {
            height: $step-line;
            top: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
            left: calc(50% + #{math.div($step-number-width, 2)});
            width: $step-line-height;
        }
    }

    &--bottom {
        flex-direction: column;
    }

    &--top {
        flex-direction: column-reverse;

        @include tracker-line() {
            top: unset;
            bottom: calc(#{math.div($step-number-width, 2)});
            left: calc(50% + #{math.div($step-number-width, 2)});
            width: $step-line-height;
        }
    }
}
```

Reply:
Nice! Will do that